provider/k8s_test.go

```diff
@@ -977,3 +1025,67 @@ func TestParquetIterator_vector32(t *testing.T) {
 		}
 	}
 }
+
+func TestBlobRegisterPrimaryFileNotFound(t *testing.T) {
+	randomVariant := uuid.New().String()
+	testResource := ResourceID{"TestBlobRegisterPrimaryFileNotFound", randomVariant, Primary}
+	testLogger := zaptest.NewLogger(t).Sugar()
+
+	tests := []struct {
+		name                 string
+		store                FileStore
+		nonExistantSourceUrl string
+		expectedErrorMessage string
+	}{
+		{"Azure", createAzureFileStore(t), "abfss://test@testingstoragegen.dfs.core.windows.net/featureform/tests/this_doesnt_exist.parquet", "source does not exist at path: featureform/tests/this_doesnt_exist.parquet"},
+		//{"S3", createS3FileStore(t), "s3://featureform-spark-testing/featureform/tests/this_doesnt_exist.parquet", "source does not exist at path: featureform/tests/this_doesnt_exist.parquet"},
```

Comment:
could not get this to properly work, need someone to look at this with me tomorrow